zulipterminal/ui_tools/views.py

```diff
@@ -1041,6 +1045,66 @@ def __init__(self, controller: Any, title: str, *, zt_version: str,
         super().__init__(controller, widgets, 'ABOUT', popup_width, title)
 
 
+class UserInfoView(PopUpView):
+    def __init__(self,
+                 controller: Any,
+                 user_id: int,
+                 title: str) -> None:
+
+        display_data = self._fetch_user_data(controller, user_id)
+
+        user_details = [(key, value) for key, value in
+                        display_data.items() if key != 'Name']
+        user_view_content = [(display_data['Name'], user_details)]
+
+        popup_width, column_widths = self.calculate_table_widths(
+            user_view_content, len(title))
+        widgets = self.make_table_with_categories(user_view_content,
+                                                  column_widths)
+
+        super().__init__(controller, widgets, 'USER_INFO', popup_width, title)
+
+    @staticmethod
+    def _fetch_user_data(controller: Any,
+                         user_id: int) -> Dict[str, Any]:
+        # Get user data from model
+        data = controller.model.get_tidied_user_info(user_id)
+
+        if not data:
+            display_data = {
+                'Name': '(Unavailable)',
+                'Error': 'User data not found',
+            }
+            return display_data
+
+        # Style the data obtained to make it displayable
+        display_data = {'Name': data['full_name']}
+
+        if data['email']:
+            display_data['Email'] = data['email']
+        if data['date_joined']:
+            display_data['Date joined'] = data['date_joined'][:10]
+        if data['timezone']:
+            display_data['Timezone'] = data['timezone']
+        if data['local_time']:
+            display_data['Local time'] = data['local_time'][11:]
+
+        if data['is_bot']:
+            display_data['Role'] = BOT_TYPE_BY_ID[data['bot_type']]
```

Comment:
To be forward-compatible, we likely want to make this into a `.get` with some kind of 'unknown' text?

Possibly the same for USER_ROLE in terms of the default value, as eg. 'moderator' would not be identified right now - which I've tested since I now am one :)

Reply:
Yeah, that makes sense. I wish to propose that as I'm returning a custom-made `role` param from `get_tidied_user_info`, I could forcefully pass `is_moderator` (with `False` as default) to it, and then handle it easily. I suppose it would make sense with regards to backward compatibility?

Noting in my above proposal, if `is_moderator` is false, it wouldn't affect the default `Member` value as well, due to the internal ordering in `role` :)